manager/assets/modext/sections/login.js

```diff
@@ -10,48 +10,81 @@ document.addEventListener('DOMContentLoaded', function() {
         loginFormUser = document.getElementById('modx-login-username'),
         resetForm = document.getElementById('modx-forgot-login-form'),
         resetFormUser = document.getElementById('modx-login-username-reset'),
+        changeLanguage = document.getElementById('modx-login-language-select'),
+        errors = document.querySelectorAll('.is-error');
+        resetFormUser = document.getElementById('modx-login-username-reset'),
         languageSelector = document.getElementById('modx-login-language-select');
 
     // When clicking on the forgot password button, swap out the forms
-    forgotPassBtn.addEventListener('click', function(e) {
-        e.preventDefault();
-        addClass(loginForm, 'is-hidden');
-        removeClass(loginForm, 'is-visible');
-        addClass(resetForm, 'is-visible');
-        removeClass(resetForm, 'is-hidden');
-        resetFormUser.focus();
-        return false;
-    });
+    if (forgotPassBtn) {
+        forgotPassBtn.addEventListener('click', function (e) {
+            e.preventDefault();
+            addClass(loginForm, 'is-hidden');
+            removeClass(loginForm, 'is-visible');
+            addClass(resetForm, 'is-visible');
+            removeClass(resetForm, 'is-hidden');
+            resetFormUser.focus();
+            removeErrors();
+            return false;
+        });
+    }
 
     // Also swap out in the reverse direction when clicking the back to login button
-    backToLoginBtn.addEventListener('click', function(e) {
-        e.preventDefault();
-        addClass(loginForm, 'is-visible');
-        removeClass(loginForm, 'is-hidden');
-        addClass(resetForm, 'is-hidden');
-        removeClass(resetForm, 'is-visible');
-        loginFormUser.focus();
-        return false;
-    });
-
-    languageSelector.addEventListener('change', function() {
-        console.log(this);
-        this.form.submit();
-    });
-
-    // Ext.get('modx-login-language-select').on('change',function(e,cb) {
-    //     var p = MODx.getURLParameters();
-    //     p.cultureKey = cb.value;
-    //     location.href = '?'+Ext.urlEncode(p);
-    // });
+    if (backToLoginBtn) {
+        backToLoginBtn.addEventListener('click', function (e) {
+            e.preventDefault();
+            addClass(loginForm, 'is-visible');
+            removeClass(loginForm, 'is-hidden');
+            addClass(resetForm, 'is-hidden');
+            removeClass(resetForm, 'is-visible');
+            loginFormUser.focus();
+            removeErrors();
+            return false;
+        });
+    }
+
+    // Change language
+    if (changeLanguage) {
```

Comment:
Any particular reason you rewrote this from the simple form submit (removed in line 37 off the diff slightly above this)? Should have the same result (GET params added while existing ones stay as-is) in a single line of easier-to-understand code...

Reply:
Hm, originally I made it to save url params, but submitting a form will leave them too. Will fix

Reply:
No, they do not added - I just double-checked and old parameters removed on submit.